src/features/auth/lib/authMachine.ts

```diff
@@ -279,21 +325,60 @@ export const authMachine = createMachine<
           ],
         },
       },
+      connectingAsGuest: {
+        entry: "setTransactionId",
+        invoke: {
+          src: async (context) => {
+            if (context.user.type !== "GUEST") throw new Error("Not a guest");
+
+            // Sleep 500ms to show Loading screen
+            await new Promise((resolve) => setTimeout(resolve, 500));
+
+            if (!context.user.guestKey) {
+              const guestKey = await createGuestAccount({
+                transactionId: context.transactionId as string,
+              });
+
+              localStorage.setItem(GUEST_KEY, guestKey);
+            }
+          },
+          onDone: {
+            target: "#authorised",
+            actions: assign({
+              user: (context) => ({
+                ...context.user,
+                guestKey: localStorage.getItem(GUEST_KEY),
+              }),
+            }),
+          },
+          onError: {
+            target: "unauthorised",
+            actions: "assignErrorMessage",
+          },
+        },
+      },
       setupContracts: {
         invoke: {
-          src: async (context, event) => {
-            const type: WalletType = (event as any).data?.wallet ?? "METAMASK";
-            await wallet.initialise(context.provider, type);
-            await communityContracts.initialise(context.provider);
+          src: async (context) => {
+            if (context.user.web3) {
```

Comment:
Would a user without a wallet enter the `setupContracts` flow?

Reply:
Hopefully not, but we need to check the context we are using has been set by a previous action.